lib/pluginmanager/util.rb

```diff
@@ -37,6 +37,36 @@ def self.logstash_plugin?(plugin, version = nil)
     end
   end
 
+  # Fetch latest version information as in rubygems
+  # @param [String] The plugin name
+  # @param [Hash] Set of available options when fetching the information
+  # @option options [Boolean] :pre Include pre release versions in the search (default: false)
+  # @return [Hash] The plugin version information as returned by rubygems
+  def self.fetch_latest_version_info(plugin, options={})
+    require "gems"
+    options = { :pre => false }.merge(options)
+    versions = Gems.versions(plugin)
+    if !options[:pre]
+      versions.select! { |version| !version["prerelease"] }
```

Comment:
This would be cleaner as

`version = versions.select { |version| !version["prerelease"] } unless @options[:pre]`

you'd have to declare `version` as null, but mutating values like this is generally confusing